recipes/default.rb

```diff
@@ -124,7 +97,53 @@
   find #{node['jmxtrans']['log_dir']} ! -name '*.gz' -mtime +2 -exec gzip '{}' \\;"
 end
 
-service 'jmxtrans' do
-  supports :restart => true, :status => true, :reload => true
+java_command = ['/usr/bin/java',
+                '-Xms${HEAP_SIZE}M',
+                '-Xmx${HEAP_SIZE}M',
+                '-XX:+UseConcMarkSweepGC',
+                '-XX:NewRatio=${NEW_RATIO}',
+                '-XX:NewSize=${NEW_SIZE}m',
+                '-XX:MaxNewSize=${NEW_SIZE}m',
+                '-XX:MaxTenuringThreshold=16',
+                '-XX:GCTimeRatio=9',
+                '-XX:PermSize=${PERM_SIZE}m',
+                '-XX:MaxPermSize=${MAX_PERM_SIZE}m',
+                '-XX:+UseTLAB',
+                '-XX:CMSInitiatingOccupancyFraction=${IO_FRACTION}',
+                '-XX:+CMSIncrementalMode',
+                '-XX:+CMSIncrementalPacing',
+                '-XX:ParallelGCThreads=${CPU_CORES}',
+                '-Dsun.rmi.dgc.server.gcInterval=28800000',
+                '-Dsun.rmi.dgc.client.gcInterval=28800000',
+                '-Djava.awt.headless=true',
+                '-Djava.net.preferIPv4Stack=true',
+                '-Djmxtrans.log.level=${LOG_LEVEL}',
+                '-Djmxtrans.log.dir=${LOG_DIR}',
+                '-jar',
+                '$JAR_FILE',
+                '-e',
+                '-j ${JSON_DIR}',
+                '-s ${SECONDS_BETWEEN_RUNS}',
+                '-c ${CONTINUE_ON_ERRORS}',
+                ].join(' ')
+
+poise_service 'jmxtrans' do
+  environment(
+    :JAR_FILE => "#{node['jmxtrans']['home']}/jmxtrans-all.jar",
```

Comment:
I suggest you use an attribute to define all these environments variables:

in attributes/default.rb
```
default['poise-service']['jmxtrans']['environment'] ={
...
}
```

poise-service cookbook is supposed to do everything else for you